src/main/java/com/vaadin/demo/component/treegrid/TreeGridRichContent.java

```diff
@@ -0,0 +1,107 @@
+import com.vaadin.flow.component.icon.VaadinIcon;
+import com.vaadin.flow.component.orderedlayout.FlexComponent;
+import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+import com.vaadin.flow.component.treegrid.TreeGrid;
+import com.vaadin.flow.router.Route;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Route("tree-grid-rich-content")
+public class TreeGridRichContent extends Div {
+
+    private List<Person> people = DataService.getPeople();
+
+    public TreeGridRichContent() {
+        TreeGrid<Person> treeGrid = new TreeGrid<>();
+        treeGrid.setItems(getManagers(), this::getStaff);
+
+        // tag::snippet[]
+        treeGrid.addComponentHierarchyColumn(person -> {
+            Avatar avatar = new Avatar();
+            avatar.setName(person.getFullName());
+            avatar.setImage(person.getPictureUrl());
+
+            Span fullName = new Span(person.getFullName());
+
+            Span profession = new Span(person.getProfession());
+            profession.getStyle()
+                    .set("color", "var(--lumo-secondary-text-color)")
+                    .set("font-size", "var(--lumo-font-size-s)");;
+
+            VerticalLayout column = new VerticalLayout(fullName, profession);
+            column.getStyle()
+                    .set("line-height", "var(--lumo-line-height-m)");
+            column.setPadding(false);
+            column.setSpacing(false);
+
+            HorizontalLayout row = new HorizontalLayout(avatar, column);
+            row.setAlignItems(FlexComponent.Alignment.CENTER);
+            row.setSpacing(true);
+            return row;
+        });
+
+        treeGrid.addComponentColumn(person -> {
+            Icon emailIcon = createIcon(VaadinIcon.ENVELOPE);
+            Span email = new Span(person.getEmail());
+
+            Anchor emailLink = new Anchor();
+            emailLink.add(emailIcon, email);
+            emailLink.setHref("mailto:" + person.getEmail());
+            emailLink.getStyle()
+                    .set("align-items", "center")
+                    .set("display", "flex");
+
+            Icon phoneIcon = createIcon(VaadinIcon.PHONE);
+            Span phone = new Span(person.getAddress().getPhone());
+
+            Anchor phoneLink = new Anchor();
+            phoneLink.add(phoneIcon, phone);
+            phoneLink.setHref("tel:" + person.getAddress().getPhone());
+            phoneLink.getStyle()
+                    .set("align-items", "center")
+                    .set("display", "flex");
+
+            VerticalLayout column = new VerticalLayout(emailLink, phoneLink);
+            column.getStyle()
+                    .set("font-size", "var(--lumo-font-size-s)")
+                    .set("line-height", "var(--lumo-line-height-m)");
+            column.setPadding(false);
+            column.setSpacing(false);
+            return column;
+        });
```

Comment:
The header is missing:
```suggestion
        }).setHeader("Contact");
```